### PR TITLE
Mcsh5 offsets and proper scaling in uV for return_scaled

### PR DESCRIPTION
**Pull request by yger**
Adding the offsets from the mcs5h header in order to have a recording with offset_to_UV and gain_to_UV. In addition, the gain was entered in V, and thus the scaling was not really making sense. As far as I understand, we want to have return_scaled returning values in uV, am I right? If yes, this is doing the job here, given the documentation that can be found online

https://www.multichannelsystems.com/sites/multichannelsystems.com/files/documents/manuals/HDF5%20MCS%20Raw%20Data%20Definition.pdf

### CODE REVIEW

File: src/spikeinterface/extractors/mcsh5extractors.py [hunk 1]
```diff
@@ -121,7 +124,8 @@ def openMCSH5File(filename, stream_id):
     Tick = info["Tick"][0] / 1e6
     exponent = info["Exponent"][0]
     convFact = info["ConversionFactor"][0]
-    gain = convFact.astype(float) * (10.0**exponent)
+    gain_uV = 1e6 * (convFact.astype(float) * (10.0**exponent))
+    offset_uV = 1e6 * (info["ADZero"].astype(float) * (10.0**exponent))
```
Comment (h-mayorquin) [inline src/spikeinterface/extractors/mcsh5extractors.py:128]: Given this from the specification:
image


Shouldn't the offset be negative?

**Review by h-mayorquin — APPROVED** [review]
This makes sense to me save the question  about the offset. Could we add a link to the specification in the docstring now that we are a it?

https://www.multichannelsystems.com/downloads/documentation?page=3

This link or something better if you know.

File: src/spikeinterface/extractors/mcsh5extractors.py (hunk 1 above)
Comment (h-mayorquin) [reply src/spikeinterface/extractors/mcsh5extractors.py]: Great.

Comment (samuelgarcia) [reply src/spikeinterface/extractors/mcsh5extractors.py]: In our implementation the offset is after the gain.
So the offset should be divided by gain I think. Can you check ?

**Comment by samuelgarcia** [comment]
Does the dtype is always a uint ? why is there this offset ?

File: src/spikeinterface/extractors/mcsh5extractors.py [hunk 2]
```diff
@@ -122,7 +129,8 @@ def openMCSH5File(filename, stream_id):
     Tick = info["Tick"][0] / 1e6
     exponent = info["Exponent"][0]
     convFact = info["ConversionFactor"][0]
-    gain = convFact.astype(float) * (10.0**exponent)
+    gain_uV = 1e6 * (convFact.astype(float) * (10.0**exponent))
+    offset_uV = -1e6 * (info["ADZero"].astype(float) * (10.0**exponent))
```
Comment (alejoe91) [inline src/spikeinterface/extractors/mcsh5extractors.py:133]:
```suggestion
    offset_uV = -1e6 * (info["ADZero"].astype(float) * (10.0**exponent)) / gain_uV
```

Comment (alejoe91) [reply src/spikeinterface/extractors/mcsh5extractors.py:133]: See @samuelgarcia commeent

File: src/spikeinterface/extractors/mcsh5extractors.py [hunk 3]
```diff
@@ -122,7 +129,8 @@ def openMCSH5File(filename, stream_id):
     Tick = info["Tick"][0] / 1e6
     exponent = info["Exponent"][0]
     convFact = info["ConversionFactor"][0]
-    gain = convFact.astype(float) * (10.0**exponent)
+    gain_uV = 1e6 * (convFact.astype(float) * (10.0**exponent))
+    offset_uV = -1e6 * (info["ADZero"].astype(float) * (10.0**exponent)) / gain_uV
```
Comment (samuelgarcia) [inline src/spikeinterface/extractors/mcsh5extractors.py:133]:
```suggestion
    offset_uV = -1e6 * (info["ADZero"].astype(float) * (10.0**exponent)) * gain_uV
```

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**